src/common.ts

```diff
@@ -10,6 +17,11 @@ export interface Config {
   };
 }
 
+export type PreHookFunction = (
+  change: DocumentSnapshot | Change<QueryDocumentSnapshot>,
+  context: functions.EventContext
+) => Promise<void>;
```

Comment:
Consider making the return type ` Promise<void> | void` that way clients dont have to use the async keyword if they usign await